### ISSUE TITLE
Add Flatcar to supported distro list

### ISSUE THREAD

**Issue by johananl** [kubernetes-sigs/cluster-api-provider-vsphere#1694]
Now that we have [Ignition support](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/1429), we can add Flatcar Container Linux to the list of distros supported by CAPV (and possibly other Ignition-powered distros, though we'd have to test them first).

We may also want to figure out how to automate building images for Flatcar (should this be a prerequisite?).

**Comment by dharmjit** [comment]
> We may also want to figure out how to automate building images for Flatcar

@johananl Do the Flatcar Linux CAPI images(even if build manually) exists somewhere?

**Comment by johananl** [comment]
@dharmjit you'd have to build your own for the time being. Use image-builder for that: https://image-builder.sigs.k8s.io/capi/providers/vsphere.html

We might start publishing official images in the future.

**Comment by srm09** [comment]
I have had some chats with the `image-builder` folks but haven't gotten any traction on setting up jobs to building image on a regular cadence. I have started to build images manually once a month and upload those to the `cape-templates` bucket. Here are the last couple of PRs I raised:
- https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/1777
- https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/1737

**Comment by srm09** [comment]
That said, if you can write a small gist on building the flatcar images using `image-builder`, I can manually also start creating these once a month too.

**Comment by johananl** [comment]
@srm09 thanks! Highly appreciated.

To build Flatcar OVAs for vSphere, run `make build-node-ova-vsphere-flatcar`. I've compiled a [gist](https://gist.github.com/johananl/89b1f73187b15c997ce967891accc252) with more detailed instructions.

Note that Flatcar has 4 [release channels](https://www.flatcar.org/releases). I suggest we use the latest `stable` release at any given time to build Flatcar images. If we had automation then we'd probably want to regularly build all channels, but this is out of scope here I guess.

Is there a reason you build images manually? Would you like some help with automating this process?

**Comment by srm09** [comment]
I have a couple of questions from the gist, 
`FLATCAR_VERSION` => what value do you use?

I would suggest going with the ` make build-node-ova-vsphere-flatcar` which am assuming uses the `stable` channel?

**Comment by johananl** [comment]
>`FLATCAR_VERSION` => what value do you use?

## TL;DR

For building images for the `capv-templates` bucket, I'd use the following command without specifying a channel or a version:

```
make build-node-ova-vsphere-flatcar
```

This would build an image using the most recent release in the `stable` channel, which makes sense for this use case.

## A longer explanation

Flatcar has 4 **channels**. Each channel has an ever-growing list of **releases**. Looking at https://www.flatcar.org/releases, at the time of writing the latest release in the `stable` channel is `3374.2.4`.

When running `make build-node-ova-vsphere-flatcar`, you can specify a channel using `FLATCAR_CHANNEL` and a release using `FLATCAR_VERSION`.

The default value of `FLATCAR_CHANNEL` is `stable`. The default value of `FLATCAR_VERSION` is `current` which always points at the most recent release in the selected channel.

**Comment by srm09** [comment]
Thanks for the detailed explanation. I was able to create a v1.25.6 ova using the first make target mentioned above.
<img width="1028" alt="Screenshot 2023-02-21 at 09 18 55" src="https://user-images.githubusercontent.com/8758225/220415357-93d76451-d8f0-4d8d-a70e-969ec30c1a01.png">

Feel free to open a PR to add the OVA link to the CAPV README

**Comment by johananl** [comment]
>Feel free to open a PR to add the OVA link to the CAPV README

Done! https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/1796
Thanks again for your consistent assistance @srm09 🙏
